api/mgo/mongo.go

```diff
@@ -126,3 +128,33 @@ func GenerateFind(condition utils.M) utils.M {
 
 	return m
 }
+
+// Profile fires a profile request
+func (s *Mongo) Profile(id string) (*model.Response, error) {
+	m := &proto.Meta{DbType: s.db, Project: s.config.Project, Token: s.config.Token}
+	return s.config.Transport.Profile(context.TODO(), m, id)
+}
+
+// Profiles fires a profiles request
+func (s *Mongo) Profiles() (*model.Response, error) {
+	m := &proto.Meta{DbType: s.db, Project: s.config.Project, Token: s.config.Token}
+	return s.config.Transport.Profiles(context.TODO(), m)
+}
+
+// SignIn fires a signIn request
+func (s *Mongo) SignIn(email, password string) (*model.Response, error) {
+	m := &proto.Meta{DbType: s.db, Project: s.config.Project, Token: s.config.Token}
+	return s.config.Transport.SignIn(context.TODO(), m, email, password)
+}
+
+// SignUp fires a signUp request
+func (s *Mongo) SignUp(email, name, password, role string) (*model.Response, error) {
+	m := &proto.Meta{DbType: s.db, Project: s.config.Project, Token: s.config.Token}
+	return s.config.Transport.SignUp(context.TODO(), m, email, name, password, role)
+}
+
+// EditProfile fires a editProfile request
+func (s *Mongo) EditProfile(id, email, name, password string) (*model.Response, error) {
```

Comment:
Shouldn't we take email, name and password as a struct. Passing it as parameters makes them look compulsory. A struct on the other hand can have fields left unset. Lets make this change for SQL as well. We should do this in python too.

Reply:
Im not sure about this though. @AliabbasMerchant what do you think?

Reply:
Even I was unsure about this.  
(In python I have kept them as optional, so no prob)  
I think a struct should be ok.

```
type new_values struct {
    email string
    name string
    password string
}
```